Add custom fields to Update-RedmineIssue

TheBakaBandit opened the pull request:
Added an option to change values in custom fields if needed

TheBakaBandit committed:
Add custom fields to Update-RedmineIssue

Added an option to change values in custom fields if needed

majkinetor commented on mm-redmine.psm1:
```diff
@@ -179,7 +179,8 @@ function Update-RedmineIssue {
         [switch] $PrivateNotes,
-        [array]  $Uploads
+        [array]  $Uploads,
+        [array]  $CustomFields
```
It can't be done half baked like this. There is also `New-RedmineIssue` function

TheBakaBandit committed:
Added Custom-Fields as HashTable

Added custom field as Hash table, the Keys are the "id"  of a custom field.
Added to both update-RedmineIssue and get-RedmineIssue.

majkinetor commented on mm-redmine.psm1:
```diff
@@ -153,6 +154,12 @@ function New-RedmineIssue {
+    if ($CustomFields) { 
+        $issue.custom_fields = @()
+        foreach($element in $CustomFields.GetEnumerator()){
+            $issue.custom_fields += ConvertTo-Json @{ id = $element.name ; value = $element.value;  }
```
ConvertTo-Json is done by send-request, this one doubles it and so it wont work.

majkinetor reviewed (changes requested):
ConvertTo-Json still present

TheBakaBandit committed:
Some more fixes

Fixed CustomFields for New-RedminIssue and Update-RedmineIssue, also made the code more consistent.
Added a check in Get-RedmineIssue as it threw a non-terminating error when calling an issue by ID only

TheBakaBandit commented:
Newest commit should fix all issues

majkinetor commented on mm-redmine.psm1:
```diff
@@ -281,7 +295,9 @@ function Get-RedmineIssue {
     $pInclude  = if ($Include) { "&include={0}" -f ($Include -join ',') }
-    if ($Id) { $pInclude = $pInclude.Replace('&', '?') }
+    if ($Id) { 
+        if ($include) {$pInclude = $pInclude.Replace('&', '?') }
```
?

majkinetor commented on mm-redmine.psm1:
```diff
@@ -281,7 +295,9 @@ function Get-RedmineIssue {
     $pInclude  = if ($Include) { "&include={0}" -f ($Include -join ',') }
-    if ($Id) { $pInclude = $pInclude.Replace('&', '?') }
+    if ($Id) { 
+        if ($include) {$pInclude = $pInclude.Replace('&', '?') }
```
Not a good form - if within if can be avoided. `$Include` is also a function param, by obvious convention in this module all function use large first letter for parameters so you can easily distinguish them from local vars in the code.

What was wrong with previous one anyway ? If `$Include` was not present it would replace nothing and we would have empty variable.

TheBakaBandit replied:
In my use case it caused me some problems in error processing, cause it throws `You cannot call a method on a null-valued expression.`  But I can revert the changes if they are not more widely applicable.

TheBakaBandit replied:
The actual code I should have written is `if ($Id -and $Include) { $pInclude = $pInclude.Replace('&', '?') }`

majkinetor replied:
Replace requires string object indeed.

majkinetor replied:
Yes.. use alternative code.

TheBakaBandit committed:
made code more consistent

majkinetor commented:
LGTM. Thanks.
